components/BillTestimonies/BillTestimonies.js

```diff
@@ -1,21 +1,28 @@
 import React from "react";
-import { testimonies } from "../MockTestimonies"
 import { Table, Container } from 'react-bootstrap'
 import ExpandTestimony from "../ExpandTestimony/ExpandTestimony";
+import { usePublishedTestimonyListing } from "../db";
+
 
 // the word "testimonies": In more general, commonly used, contexts, the plural form will also be testimony.  However, in more specific contexts, the plural form can also be testimonies e.g. in reference to various types of testimonies or a collection of testimonies.
 
+
 const ViewTestimonies = (props) => {
   const bill = props.bill
-  const testimoniesComponent = !testimonies ? "" :
+  const testimoniesResponse = usePublishedTestimonyListing({billId: bill.id})
+
+  const testimonies = testimoniesResponse.status == "loading" ? [] : testimoniesResponse.result
+
+  const testimoniesComponent = testimonies.loading ? "" :
     testimonies.map((testimony, index) => {
-      if (testimony.billNumber === bill.BillNumber) {
+      // need to get the author email address (unless it is anonymous)
+      if (testimony.billId === bill.BillNumber) {
```

Comment:
I think we can remove this if-else since we specify athe billId in the hook, so all testimony should be for the given bill.

Reply:
yes, good catch.

Reply:
I also had to change the hook call from 
const testimoniesResponse = usePublishedTestimonyListing({billId: bill.id})
to
const testimoniesResponse = usePublishedTestimonyListing({billId: bill.BillNumber})